### PR TITLE
Implement whitening

### PR DESCRIPTION
**Pull request by husxk**
I am merging this at 10:30

### CODE REVIEW

**Comment by husxk** [comment]
@kornelu123 can you look at this?

File: data_process.hpp [hunk 1]
```diff
@@ -12,12 +12,43 @@ class data_process
...
+    uint8_t get_bit(uint8_t x)
+    {
+      return as_unsigned & (1 << x);
+    }
+
+    void cycle()
+    {
+      as_unsigned <<= 1;
+
+      uint8_t msb_bit = as_unsigned >> 7;
```
Comment (kornelu123) [inline data_process.hpp]: `const` ?

**Review by kornelu123 — COMMENTED** [review]
seems fine, one comment

File: data_process.hpp (hunk 1 above)
Comment (husxk) [reply data_process.hpp]: yup, it should be const, will  be changed

**Pull request merged**